game_event/views.py

```diff
@@ -150,3 +154,32 @@ def process_answer_game_event(request, id):
     content['event_exists'] = True
 
     return render(request, 'game_event/result.html', content)
+
+
+def get_weather_from_api(game_events, weather_getter):
+    for game in game_events:
+        url = 'https://api.openweathermap.org/data/2.5/weather'
```

Comment:
no reason to create the same URL each iteration.
```suggestion
    url = 'https://api.openweathermap.org/data/2.5/weather'
    for game in game_events:
        
```


    
also true for the `params `variable
You can define the dictionary with the common values, `units `and `appid`, before the for loop since they are the same for all game_events. You can change the variable `params['q']` inside the loop based on the specific game event's `court.city`."

This change improves the readability of the function

Reply:
We took out the constant `params ` - `units `, `url  `and `appid `out of the for loop.